Inductor cpp wrapper: support LinearBinary

chunyuan-w opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #99957

Support `mkldnn::_linear_pointwise.binary` in cpp wrapper for GPT-J.


cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

chunyuan-w committed:
Inductor cpp wrapper: support LinearBinary

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support LinearBinary"


Support `mkldnn::_linear_pointwise.binary` for GPT-J.


cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support LinearBinary"


Support `mkldnn::_linear_pointwise.binary` for GPT-J.


cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support LinearBinary"


Support `mkldnn::_linear_pointwise.binary` for GPT-J.


cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support LinearBinary"


Support `mkldnn::_linear_pointwise.binary` for GPT-J.


cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support LinearBinary"


Support `mkldnn::_linear_pointwise.binary` in cpp wrapper for GPT-J.


cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support LinearBinary"


Support `mkldnn::_linear_pointwise.binary` in cpp wrapper for GPT-J.


cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

EikanWang reviewed (approved):
LGTM. Meanwhile, please check if the code should be moved to mkldnn.py.

chunyuan-w commented:
> LGTM. Meanwhile, please check if the code should be moved to mkldnn.py.

I checked `mkldnn.py` and it only contains mkldnn related `Module`s while all the mkldnn related `ir`s are still in `ir.py`. cc @XiaobingSuper

chunyuan-w committed:
Update on "Inductor cpp wrapper: support LinearBinary"


Support `mkldnn::_linear_pointwise.binary` in cpp wrapper for GPT-J.


cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w commented on torch/_inductor/graph.py:
```diff
@@ -75,8 +75,11 @@ def supported_dtype_of_cpp_wrapper(dtype, cuda):
+    ), "get_constant_buffer_dtype only supports input of sympy.Symbol or sympy.core.numbers.Integer"
+    if isinstance(constant_buffer, sympy.core.numbers.Integer):
+        return torch.int64
+    assert isinstance(constant_buffer, sympy.Symbol)
```
Removed the redundant assert.

chunyuan-w commented:
@pytorchbot merge
